lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable no-use-before-define */
+
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+// NOTE --> Keep this file in sync with ../stage.js
+
+// NOTE: This code was written since there are problem setting up dedicated CloudFormation
+// Stage resource (see https://github.com/serverless/serverless/pull/5692#issuecomment-467849311 for more information).
+
+module.exports = {
+  updateStage() {
+    // this array is used to gather all the patch operations we need to
+    // perform on the stage
+    this.apiGatewayStagePatchOperations = [];
+    this.apiGatewayStageState = null;
+
+    return BbPromise.bind(this)
+      .then(getAccountId)
+      .then(getRestApiId)
+      .then(getStage)
+      .then(handleTracing)
+      .then(handleLogs)
+      .then(handleTags)
+      .then(applyUpdates);
+  },
+};
+
+function getAccountId() {
+  return this.provider.getAccountId().then((id) => this.accountId = id); // eslint-disable-line
+}
+
+function getRestApiId() {
```

Comment:
Conceptually it's more `resolveRestApiId` (from `getRestApiId` we rather expect to return the id).

Similarly with other `get*` methods here

Reply:
Good idea! :+1: